app/controllers/npcs_controller.rb

```diff
@@ -13,15 +13,31 @@ def create
         end
       end
 
-      private
-      def npc_params
-        params.require(:npc).permit(:name, :address, :likes, :image)
-      end
+      
   
     def update
+      npc = Npc.find(params[:id])
+      npc.update(npc_params)
+      if npc.valid?
+        render json: npc
+      else
+        render json: npc.errors, status: 422
+      end
     end
   
     def destroy
+      npc = Npc.find(params[:id])
+      npc.destroy(npc_params)
+      if npc.destroy
```

Comment:
You don't need line 30 if you have line 31 here.

Reply:
Had to do some stuff on the frontend to get it work